# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1468431396 14400
#      Wed Jul 13 13:36:36 2016 -0400
# Node ID 4e770056f74ad27ff900fb94fc9acea3859f1c38
# Parent  67a5341edf15f0e448c8b52641cc93039c413c10
don't attempt to access intepreter before it is ready (bug #48494)

* octave.cc (application::interactive): Avoid dereferencing
m_interpreter if it is 0.

diff --git a/libinterp/octave.cc b/libinterp/octave.cc
--- a/libinterp/octave.cc
+++ b/libinterp/octave.cc
@@ -339,27 +339,32 @@ namespace octave
       }
 
     symbol_table::assign (".nargin.", nargs);
     symbol_table::mark_hidden (".nargin.");
   }
 
   void application::interactive (bool arg)
   {
-    m_interpreter->interactive (arg);
+    interpreter *interp = instance->m_interpreter;
+
+    if (interp)
+      interp->interactive (arg);
   }
 
   bool application::forced_interactive (void)
   {
     return instance->m_options.forced_interactive ();
   }
 
   bool application::interactive (void)
   {
-    return instance->m_interpreter->interactive ();
+    interpreter *interp = instance->m_interpreter;
+
+    return interp ? interp->interactive () : false;
   }
 
   application::~application (void)
   {
     instance = 0;
 
     delete m_interpreter;
   }
